tlr/stages-tlr/model-provider/src/main/java/edu/kit/kastel/mcse/ardoco/tlr/models/informants/LargeLanguageModel.java

```diff
@@ -0,0 +1,104 @@
+    CODELLAMA_13B("Codellama 13b", () -> createOllamaModel("codellama:13b")), //
+    // CODELLAMA_70B("Codellama 70b", () -> createOllamaModel("codellama:70b")), //
+    //
+    // GEMMA_2_27B("Gemma2 27b", () -> createOllamaModel("gemma2:27b")), //
+    //
+    // QWEN_2_72B("Qwen2 72b", () -> createOllamaModel("qwen2:72b")), //
+    //
+    LLAMA_3_1_8B("Llama3.1 8b", () -> createOllamaModel("llama3.1:8b-instruct-fp16")), //
+    LLAMA_3_1_70B("Llama3.1 70b", () -> createOllamaModel("llama3.1:70b")), //
+    //
+    // MISTRAL_7B("Mistral 7b", () -> createOllamaModel("mistral:7b")), //
+    // MISTRAL_NEMO_27B("Mistral Nemo 12b", () -> createOllamaModel("mistral-nemo:12b")), //
+    // MIXTRAL_8_X_22B("Mixtral 8x22b", () -> createOllamaModel("mixtral:8x22b")), //
+    //
+    // PHI_3_14B("Phi3 14b", () -> createOllamaModel("phi3:14b")), //
+    //
+    OLLAMA_GENERIC(Environment.getenv("OLLAMA_MODEL_NAME"), () -> createOllamaModel(Environment.getenv("OLLAMA_MODEL_NAME")));
+
+    private final Supplier<ChatModel> creator;
+    private final String humanReadableName;
+
+    LargeLanguageModel(String humanReadableName, Supplier<ChatModel> creator) {
+        this.humanReadableName = humanReadableName;
+        this.creator = creator;
+    }
+
+    public String getHumanReadableName() {
+        return humanReadableName;
+    }
+
+    public ChatModel create() {
+        return new CachedChatLanguageModel(creator.get(), this.name());
+    }
+
+    public boolean isGeneric() {
+        return this.name().endsWith("_GENERIC");
+    }
+
+    public boolean isOpenAi() {
+        return this.name().startsWith("GPT_");
+    }
+
+    private static final int SEED = 422413373;
+
+    /**
+     * Creates an OpenAI chat model instance.
+     * Requires OpenAI organization ID and API key to be set in environment variables.
+     *
+     * @param model The name of the model to use
+     * @return A configured OpenAI chat model instance
+     * @throws IllegalStateException If required environment variables are not set
+     */
+    private static OpenAiChatModel createOpenAiModel(String model) {
+        String openAiOrganizationId = Environment.getenv("OPENAI_ORGANIZATION_ID");
+        String openAiApiKey = Environment.getenv("OPENAI_API_KEY");
+        if (openAiOrganizationId == null || openAiApiKey == null) {
+            throw new IllegalStateException("OPENAI_ORGANIZATION_ID or OPENAI_API_KEY environment variable not set");
+        }
+        return new OpenAiChatModel.OpenAiChatModelBuilder().modelName(model)
+                .organizationId(openAiOrganizationId)
+                .apiKey(openAiApiKey)
+                .temperature(0.0)
+                .seed(SEED)
+                .build();
+    }
+
+    private static OllamaChatModel createOllamaModel(String model) {
+        String host = Environment.getenv("OLLAMA_HOST");
+        String user = Environment.getenv("OLLAMA_USER");
+        String password = Environment.getenv("OLLAMA_PASSWORD");
```

Comment:
What if the ollama instance does not use username and password (maybe API-key instead)?